src/test/java/com/parasoft/findings/utils/results/testableinput/FindingsLocationMatcherTest.java

```diff
@@ -0,0 +1,158 @@
+
+        assertEquals(file.getAbsolutePath(), result);
+    }
+
+    @Test
+    public void testGetFilePath_fileNotExist_FileTestableInput() {
+        File file = new File("test/file");
+        FileTestableInput fileTestableInput = new FileTestableInput(file);
+
+        String result = FindingsLocationMatcher.getFilePath(fileTestableInput);
+
+        assertEquals(file.getAbsolutePath(), result);
+    }
+
+    @Test
+    public void testGetFilePath_fileNotExist_ProjectFileTestableInput() {
+        File file = new File("test/file");
+        ProjectFileTestableInput projectFileTestableinput = new ProjectFileTestableInput(file,
+                "project", "project", "/project", "relative/path");
+
+        String result = FindingsLocationMatcher.getFilePath(projectFileTestableinput);
+
+        assertEquals("project/relative/path", result);
+    }
+
+    @Test
+    public void testCompute_onlyProjectNameInProjectPath() throws LocationsException {
+        PathInput pathInput = new PathInput("/projectName");
+
+        ITestableInput result = findingsLocationMatcher.matchLocation(pathInput, Collections.singletonList(0L), null, null, false);
+
+        assertEquals("projectName", result.getName());
+
+        // Verify override functions in FileTestableInput
+        FileTestableInput fileTestableInput = new FileTestableInput(new File("/projectName"));
+
+        assertEquals(fileTestableInput.toString(), result.toString());
+        assertEquals(fileTestableInput.hashCode(), result.hashCode());
+        // When compared object is not FileTestableInput type
+        assertFalse(result.equals("test"));
+        // When compared object is same as result
+        assertTrue(result.equals(fileTestableInput));
+    }
+
+    @Test
+    public void testCompute_oneSegmentInProjectPath() throws LocationsException {
+        PathInput pathInput = new PathInput("test/project");
+
+        ITestableInput result = findingsLocationMatcher.matchLocation(pathInput, Collections.singletonList(0L), null, null, false);
+
+        assertEquals("test", ((ProjectFileTestableInput) result).getProjectName());
+        assertEquals("/test", ((ProjectFileTestableInput) result).getProjectPath());
+        assertEquals("project", ((ProjectFileTestableInput) result).getProjectRelativePath());
+
+        // Verify override functions in ProjectFileTestableInput
+        ProjectFileTestableInput projectFileTestableInput = new ProjectFileTestableInput(new File("test/project"),
+                "test", "project", "/project", "project");
+
+        assertEquals(projectFileTestableInput.hashCode(), result.hashCode());
+        // When compared object is not ProjectFileTestableInput type
+        assertFalse(result.equals("test"));
+        // When compared object is ProjectFileTestableInput type
+        assertTrue(result.equals(projectFileTestableInput));
+        // When compared object is same as result
+        assertTrue(result.equals(result));
```

Comment:
Why do we need this assertion? It doesn't make sense to me to compare the same object.